### DISCUSSION TITLE
Why duckdb uses page_offset and page_nr in AggregaHashTable instead of pointer?

### DISCUSSION THREAD

**Discussion by YjyJeff** [duckdb/duckdb#3856] [Q&A] [answered]
I am interested in how duckdb works. After reading the [Parallel Grouped Aggregation in Duckdb](https://duckdb.org/2022/03/07/aggregate-hashtable.html), it says 

> The pointers are not actual pointers but symbolic, they refer to a block ID and a row offset within said block

Why not use the pointer here? I know the pointer can be constructed from block ID and row offset, but it needs to do arithmetic. What are the advantages of these two variables? 

Thanks in advance!

**Comment by hannes** [comment, marked as answer]
If those are not pointers the payload blocks can be moved to disk temporarily to reduce memory pressure. This is not done yet but planned.
